client/my-sites/stats/site.jsx

```diff
@@ -918,7 +926,8 @@ export default connect(
 			supportUserFeedback,
 		} = getEnvStatsFeatureSupportChecks( state, siteId );
 
-		const hasSiteLoadedFeatures = hasLoadedSiteFeatures( state, siteId );
+		// Odyssey Stats does not need loading features to determine gated features.
+		const hasSiteLoadedFeatures = hasLoadedSiteFeatures( state, siteId ) || isOdyssey;
```

Comment:
```suggestion
		const hasSiteLoadedFeatures = isOdyssey || hasLoadedSiteFeatures( state, siteId );
```

If we swap these conditions, we'll avoid calling the function at all inside Odyssey.